support showboard again, optionally enablable with --showboard

wonderingabout opened the pull request:
@roy7 @Dorus @windo 

general look here : https://github.com/wonderingabout/gtp2ogs/tree/config-showboard

showboard support was removed in :
https://github.com/online-go/gtp2ogs/commit/d5ebdbbde259a97c5ae1aed0ec42a07c9fbb2dbf

this was because some bots dont support it and it breaks them (for example
PhoenixGo),

adding optional support for it again : 
to enable support for GTP command showboard, use the
gtp2ogs argument `--showboard` (which is config.SHOWBOARD)

also, some bot admins may want to disable it for smaller
and easier to read log, so giving the option to enable it or 
not is also useful in that case

so supporting this again but default is disabled

didnt test it (my main bot is PhoenixGo), but i hope it 
should work without any issue

windo commented on config.js:
```diff
@@ -64,6 +65,7 @@ exports.updateFromArgv = function() {
         .describe('logfile', 'In addition to logging to the console, also log gtp2ogs output to a text file')
         .describe('json', 'Send and receive GTP commands in a JSON encoded format')
         .describe('kgstime', 'Set this if bot understands the kgs-time_settings command')
+        .describe('showboard', 'Set this if bot understands the showboard GTP command and you want to display the GTP showboard output')
```
The description could be shortened to "Set this to issue showboard GTP command on every move".

wonderingabout replied:
@windo 

while your sentence is shorter, it is more complicated to understand (for example the word "issue" does not have a simple meaning here)

i wanted to highlight that even if the bot supports showboard command, bot admin can still disabled it if he wants

i think we can shorten it like this (we lose a bit of clear sense, but it should be easy to guess, and can be detailed in OPTIONS-list)

`and you want to display it`

wonderingabout replied:
because by `display it` it may mean : 
- display the GTP command text "showboard"
- or display the GTP showboard output

so i'll try to think a little more if there a short and clear way to write it

wonderingabout replied:
also we may add a coma at the middle of the sentence, and use "and if" to make it clearer too

wonderingabout commented:
removed the word GTP in 2nd half of sentences
added a coma + "and if" to make it even clearer
slightly fixed commit message

so a bit less repetitive, but about same length length
and overall more satisfying to me :+1: 

feel free to add other suggestions :+1:

wonderingabout commented:
please dont merge yet, i'll remove the ugly merge commit first lol

wonderingabout commented:
@roy7 
ready to merge again :+1: 

will now commit the `===` changes again on top of this
the `let i of` will be done either another time, or when we add functions instead

wonderingabout commented:
@windo 

when testing that it behaved very nicely, PhoenixGo was killed after its crash, and after 5 restart kills it resigned, see log below 

but i think we should add a sendchat abotu why the bot resigns, or it's confusing for the user, i'll commit it :+1: 

```
You are using gtp2ogs version 6.0
- For changelog or latest devel updates, please visit https://github.com/online-go/gtp2ogs/tree/devel

Warning: /--komi auto/ has been renamed to /--komi automatic/



Warning : No nopause setting detected, games are likely to last forever


Feb 24 20:38:17   #  Connecting to https://beta.online-go.com:443
Feb 24 20:38:18   #  Connected
Feb 24 20:38:18   #  Bot is username: meta--金毛测试-20b
Feb 24 20:38:18   #  Bot is user id: 787
Feb 24 20:38:26   #  Accepting challenge from metaphysician splurgist :) (13k)  [19x19]  id = 5073
Feb 24 20:38:27   [Game 5073]            Connecting to game.
Feb 24 20:38:27   [Game 5073]            gamedata     W   metaphysician splurgist :)  [19x19]    
Feb 24 20:38:27   [Game 5073]            gamedata     W   metaphysician splurgist :)  [19x19]    
Feb 24 20:38:46   [Game 5073]  Move 1    Got     q16
Feb 24 20:38:46   [Game 5073]  Move 1    Starting new bot process [13477]
Feb 24 20:38:46   [Game 5073]  Move 1    State loading for new bot
Feb 24 20:38:46   [Game 5073]  Move 1    genmove white
Feb 24 20:38:46 !!!!! [13477] stderr: I0224 20:38:46.504191 13477 mcts_main.cc:75] load config succ: 
num_eval_threads: 1
num_search_threads: 8
max_search_tree_size: 400000000
max_children_per_node: 64
max_simulations_per_step: 3000
eval_batch_size: 4
eval_wait_batch_timeout_us: 100
model_config {
  train_dir: "/home/amd2018/PhoenixGo/ckpt/"
  enable_tensorrt: true
  tensorrt_model_path: "/home/amd2018/PhoenixGo/ckpt/zero.ckpt-20b-v1.FP32.PLAN"
}
gpu_list: "0"
c_puct: 2.5
virtual_loss: 1
enable_resign: true
v_resign: -0.8
dirichlet_noise_alpha: 0.03
dirichlet_noise_ratio: 0.25
policy_temperature: 0.67
early_stop {
  check_every_ms: 100
  sims_factor: 1
  sims_threshold: 2000
}
unstable_overtime {
  time_factor: 0.3
}
behind_overtime {
  time_factor: 0.3
}
time_control {
  c_denom: 20
  c_maxply: 40
  reserved_time: 1
}

Feb 24 20:38:46 !!!!! [13477] stderr: I0224 20:38:46.504771 13477 mcts_engine.cc:85] MCTSEngine: waiting all eval threads init

Feb 24 20:38:46 !!!!! [13477] stderr: I0224 20:38:46.673152 13478 trt_zero_model.cc:41] Added linear block of size 1478656
I0224 20:38:46.673178 13478 trt_zero_model.cc:41] Added linear block of size 1478656
I0224 20:38:46.673182 13478 trt_zero_model.cc:41] Added linear block of size 1478656
I0224 20:38:46.673185 13478 trt_zero_model.cc:41] Added linear block of size 1478656

Feb 24 20:38:47 !!!!! [13477] stderr: I0224 20:38:47.806629 13478 trt_zero_model.cc:97] tensorrt max batch size: 4
I0224 20:38:47.806663 13478 trt_zero_model.cc:108] tensorrt binding: inputs (19, 19, 17)
I0224 20:38:47.806682 13478 trt_zero_model.cc:108] tensorrt binding: policy (362, 1, 1)

Feb 24 20:38:47 !!!!! [13477] stderr: I0224 20:38:47.806702 13478 trt_zero_model.cc:108] tensorrt binding: value (1, 1, 1)

Feb 24 20:38:47 !!!!! [13477] stderr: I0224 20:38:47.814841 13478 mcts_engine.cc:359] EvalRoutine: init model done, global_step=639200

Feb 24 20:38:47 !!!!! [13477] stderr: I0224 20:38:47.814908 13477 mcts_engine.cc:87] MCTSEngine: all eval threads init done
I0224 20:38:47.814992 13477 mcts_main.cc:277] input cmd: boardsize 19
I0224 20:38:47.815071 13477 mcts_main.cc:286] = 
I0224 20:38:47.815114 13477 mcts_main.cc:277] input cmd: clear_board
I0224 20:38:47.815101 13487 mcts_engine.cc:922] DeleteRoutine: deleted 1 nodes, cost 0.00151ms
I0224 20:38:47.815140 13477 mcts_main.cc:286] = 
I0224 20:38:47.815156 13487 mcts_engine.cc:922] DeleteRoutine: deleted 1 nodes, cost 0.00038ms
I0224 20:38:47.815188 13477 mcts_main.cc:277] input cmd: komi 7.5

Feb 24 20:38:47 !!!!! [13477] stderr: I0224 20:38:47.815225 13477 mcts_main.cc:286] = 
I0224 20:38:47.815243 13477 mcts_main.cc:277] input cmd: play black q16
I0224 20:38:47.815480 13477 mcts_main.cc:87] Config hasn't changed
I0224 20:38:47.815527 13477 mcts_engine.cc:152] Move: pp
I0224 20:38:47.815552 13487 mcts_engine.cc:922] DeleteRoutine: deleted 1 nodes, cost 0.00043ms
I0224 20:38:47.815560 13477 mcts_engine.cc:158] 1th move(b): pp, winrate=-nan%, N=0, Q=-nan, p=0.000000, v=nan, global_step=639200
1th move(b): pp, winrate=-nan%, N=0, Q=-nan, p=0.000000, v=nan, global_step=639200
I0224 20:38:47.815594 13477 mcts_main.cc:286] = 
I0224 20:38:47.815608 13477 mcts_main.cc:277] input cmd: showboard
E0224 20:38:47.815618 13477 mcts_main.cc:262] invalid op: showboard
I0224 20:38:47.815623 13477 mcts_main.cc:286] ? unknown command
I0224 20:38:47.815634 13477 mcts_main.cc:277] input cmd: genmove white
I0224 20:38:47.815778 13477 mcts_main.cc:87] Config hasn't changed

Feb 24 20:38:47   [13477] ? unknown command
Feb 24 20:38:47   [13477] 
Feb 24 20:38:47   [13477] Stopping bot
Feb 24 20:38:47   [Game 5073]  Move 1    gamedata     W   metaphysician splurgist :)  [19x19]    
Feb 24 20:38:47   [Game 5073]  Move 1    Starting new bot process [13494]
Feb 24 20:38:47   [Game 5073]  Move 1    State loading for new bot
Feb 24 20:38:47   [Game 5073]  Move 1    genmove white
Feb 24 20:38:48 !!!!! [13494] stderr: I0224 20:38:48.034371 13494 mcts_main.cc:75] load config succ: 
num_eval_threads: 1
num_search_threads: 8
max_search_tree_size: 400000000
max_children_per_node: 64
max_simulations_per_step: 3000
eval_batch_size: 4
eval_wait_batch_timeout_us: 100
model_config {
  train_dir: "/home/amd2018/PhoenixGo/ckpt/"
  enable_tensorrt: true
  tensorrt_model_path: "/home/amd2018/PhoenixGo/ckpt/zero.ckpt-20b-v1.FP32.PLAN"
}
gpu_list: "0"
c_puct: 2.5
virtual_loss: 1
enable_resign: true
v_resign: -0.8
dirichlet_noise_alpha: 0.03
dirichlet_noise_ratio: 0.25
policy_temperature: 0.67
early_stop {
  check_every_ms: 100
  sims_factor: 1
  sims_threshold: 2000
}
unstable_overtime {
  time_factor: 0.3
}
behind_overtime {
  time_factor: 0.3
}
time_control {
  c_denom: 20
  c_maxply: 40
  reserved_time: 1
}

Feb 24 20:38:48 !!!!! [13494] stderr: I0224 20:38:48.035101 13494 mcts_engine.cc:85] MCTSEngine: waiting all eval threads init

Feb 24 20:38:48 !!!!! [13494] stderr: I0224 20:38:48.251190 13495 trt_zero_model.cc:41] Added linear block of size 1478656
I0224 20:38:48.251212 13495 trt_zero_model.cc:41] Added linear block of size 1478656
I0224 20:38:48.251216 13495 trt_zero_model.cc:41] Added linear block of size 1478656
I0224 20:38:48.251219 13495 trt_zero_model.cc:41] Added linear block of size 1478656

Feb 24 20:38:49 !!!!! [13494] stderr: I0224 20:38:49.360833 13495 trt_zero_model.cc:97] tensorrt max batch size: 4
I0224 20:38:49.360864 13495 trt_zero_model.cc:108] tensorrt binding: inputs (19, 19, 17)
I0224 20:38:49.360880 13495 trt_zero_model.cc:108] tensorrt binding: policy (362, 1, 1)
I0224 20:38:49.360894 13495 trt_zero_model.cc:108] tensorrt binding: value (1, 1, 1)

Feb 24 20:38:49 !!!!! [13494] stderr: I0224 20:38:49.368893 13495 mcts_engine.cc:359] EvalRoutine: init model done, global_step=639200

Feb 24 20:38:49 !!!!! [13494] stderr: I0224 20:38:49.368963 13494 mcts_engine.cc:87] MCTSEngine: all eval threads init done
I0224 20:38:49.369043 13494 mcts_main.cc:277] input cmd: boardsize 19
I0224 20:38:49.369104 13494 mcts_main.cc:286] = 
I0224 20:38:49.369174 13494 mcts_main.cc:277] input cmd: clear_board
I0224 20:38:49.369195 13504 mcts_engine.cc:922] DeleteRoutine: deleted 1 nodes, cost 0.00186ms
I0224 20:38:49.369223 13494 mcts_main.cc:286] = 
I0224 20:38:49.369240 13504 mcts_engine.cc:922] DeleteRoutine: deleted 1 nodes, cost 0.0004ms
I0224 20:38:49.369266 13494 mcts_main.cc:277] input cmd: komi 7.5

Feb 24 20:38:49 !!!!! [13494] stderr: I0224 20:38:49.369303 13494 mcts_main.cc:286] = 
I0224 20:38:49.369316 13494 mcts_main.cc:277] input cmd: play black q16
I0224 20:38:49.369571 13494 mcts_main.cc:87] Config hasn't changed
I0224 20:38:49.369616 13494 mcts_engine.cc:152] Move: pp

Feb 24 20:38:49 !!!!! [13494] stderr: I0224 20:38:49.369645 13494 mcts_engine.cc:158] 1th move(b): pp, winrate=-nan%, N=0, Q=-nan, p=0.000000, v=nan, global_step=639200
I0224 20:38:49.369639 13504 mcts_engine.cc:922] DeleteRoutine: deleted 1 nodes, cost 0.00088ms
1th move(b): pp, winrate=-nan%, N=0, Q=-nan, p=0.000000, v=nan, global_step=639200
I0224 20:38:49.369696 13494 mcts_main.cc:286] = 
I0224 20:38:49.369709 13494 mcts_main.cc:277] input cmd: showboard
E0224 20:38:49.369719 13494 mcts_main.cc:262] invalid op: showboard
I0224 20:38:49.369724 13494 mcts_main.cc:286] ? unknown command
I0224 20:38:49.369735 13494 mcts_main.cc:277] input cmd: genmove white
I0224 20:38:49.369876 13494 mcts_main.cc:87] Config hasn't changed

Feb 24 20:38:49   [13494] ? unknown command
Feb 24 20:38:49   [13494] 
Feb 24 20:38:49   [13494] Stopping bot
Feb 24 20:38:49   [Game 5073]  Move 1    gamedata     W   metaphysician splurgist :)  [19x19]    
Feb 24 20:38:49   [Game 5073]  Move 1    Starting new bot process [13513]
Feb 24 20:38:49   [Game 5073]  Move 1    State loading for new bot
Feb 24 20:38:49   [Game 5073]  Move 1    genmove white
Feb 24 20:38:49 !!!!! [13513] stderr: I0224 20:38:49.556747 13513 mcts_main.cc:75] load config succ: 
num_eval_threads: 1
num_search_threads: 8
max_search_tree_size: 400000000
max_children_per_node: 64
max_simulations_per_step: 3000
eval_batch_size: 4
eval_wait_batch_timeout_us: 100
model_config {
  train_dir: "/home/amd2018/PhoenixGo/ckpt/"
  enable_tensorrt: true
  tensorrt_model_path: "/home/amd2018/PhoenixGo/ckpt/zero.ckpt-20b-v1.FP32.PLAN"
}
gpu_list: "0"
c_puct: 2.5
virtual_loss: 1
enable_resign: true
v_resign: -0.8
dirichlet_noise_alpha: 0.03
dirichlet_noise_ratio: 0.25
policy_temperature: 0.67
early_stop {
  check_every_ms: 100
  sims_factor: 1
  sims_threshold: 2000
}
unstable_overtime {
  time_factor: 0.3
}
behind_overtime {
  time_factor: 0.3
}
time_control {
  c_denom: 20
  c_maxply: 40
  reserved_time: 1
}

Feb 24 20:38:49 !!!!! [13513] stderr: I0224 20:38:49.557564 13513 mcts_engine.cc:85] MCTSEngine: waiting all eval threads init

Feb 24 20:38:49 !!!!! [13513] stderr: I0224 20:38:49.715088 13514 trt_zero_model.cc:41] Added linear block of size 1478656
I0224 20:38:49.715113 13514 trt_zero_model.cc:41] Added linear block of size 1478656
I0224 20:38:49.715116 13514 trt_zero_model.cc:41] Added linear block of size 1478656
I0224 20:38:49.715121 13514 trt_zero_model.cc:41] Added linear block of size 1478656

Feb 24 20:38:50 !!!!! [13513] stderr: I0224 20:38:50.817462 13514 trt_zero_model.cc:97] tensorrt max batch size: 4
I0224 20:38:50.817494 13514 trt_zero_model.cc:108] tensorrt binding: inputs (19, 19, 17)
I0224 20:38:50.817509 13514 trt_zero_model.cc:108] tensorrt binding: policy (362, 1, 1)
I0224 20:38:50.817523 13514 trt_zero_model.cc:108] tensorrt binding: value (1, 1, 1)

Feb 24 20:38:50 !!!!! [13513] stderr: I0224 20:38:50.826591 13514 mcts_engine.cc:359] EvalRoutine: init model done, global_step=639200

Feb 24 20:38:50 !!!!! [13513] stderr: I0224 20:38:50.826653 13513 mcts_engine.cc:87] MCTSEngine: all eval threads init done
I0224 20:38:50.826740 13513 mcts_main.cc:277] input cmd: boardsize 19
I0224 20:38:50.826802 13513 mcts_main.cc:286] = 
I0224 20:38:50.826869 13513 mcts_main.cc:277] input cmd: clear_board
I0224 20:38:50.826894 13513 mcts_main.cc:286] = 
I0224 20:38:50.826875 13523 mcts_engine.cc:922] DeleteRoutine: deleted 1 nodes, cost 0.00163ms
I0224 20:38:50.826905 13513 mcts_main.cc:277] input cmd: komi 7.5
I0224 20:38:50.826917 13523 mcts_engine.cc:922] DeleteRoutine: deleted 1 nodes, cost 0.00044ms
I0224 20:38:50.826948 13513 mcts_main.cc:286] = 
I0224 20:38:50.826966 13513 mcts_main.cc:277] input cmd: play black q16

Feb 24 20:38:50 !!!!! [13513] stderr: I0224 20:38:50.827210 13513 mcts_main.cc:87] Config hasn't changed
I0224 20:38:50.827262 13513 mcts_engine.cc:152] Move: pp
I0224 20:38:50.827283 13523 mcts_engine.cc:922] DeleteRoutine: deleted 1 nodes, cost 0.00033ms
I0224 20:38:50.827291 13513 mcts_engine.cc:158] 1th move(b): pp, winrate=-nan%, N=0, Q=-nan, p=0.000000, v=nan, global_step=639200
1th move(b): pp, winrate=-nan%, N=0, Q=-nan, p=0.000000, v=nan, global_step=639200
I0224 20:38:50.827316 13513 mcts_main.cc:286] = 
I0224 20:38:50.827328 13513 mcts_main.cc:277] input cmd: showboard
E0224 20:38:50.827338 13513 mcts_main.cc:262] invalid op: showboard
I0224 20:38:50.827343 13513 mcts_main.cc:286] ? unknown command
I0224 20:38:50.827354 13513 mcts_main.cc:277] input cmd: genmove white

Feb 24 20:38:50   [13513] ? unknown command
Feb 24 20:38:50   [13513] 
Feb 24 20:38:50   [13513] Stopping bot
Feb 24 20:38:50   [Game 5073]  Move 1    gamedata     W   metaphysician splurgist :)  [19x19]    
Feb 24 20:38:50   [Game 5073]  Move 1    Starting new bot process [13530]
Feb 24 20:38:50   [Game 5073]  Move 1    State loading for new bot
Feb 24 20:38:50   [Game 5073]  Move 1    genmove white
Feb 24 20:38:51 !!!!! [13530] stderr: I0224 20:38:51.013875 13530 mcts_main.cc:75] load config succ: 
num_eval_threads: 1
num_search_threads: 8
max_search_tree_size: 400000000
max_children_per_node: 64
max_simulations_per_step: 3000
eval_batch_size: 4
eval_wait_batch_timeout_us: 100
model_config {
  train_dir: "/home/amd2018/PhoenixGo/ckpt/"
  enable_tensorrt: true
  tensorrt_model_path: "/home/amd2018/PhoenixGo/ckpt/zero.ckpt-20b-v1.FP32.PLAN"
}
gpu_list: "0"
c_puct: 2.5
virtual_loss: 1
enable_resign: true
v_resign: -0.8
dirichlet_noise_alpha: 0.03
dirichlet_noise_ratio: 0.25
policy_temperature: 0.67
early_stop {
  check_every_ms: 100
  sims_factor: 1
  sims_threshold: 2000
}
unstable_overtime {
  time_factor: 0.3
}
behind_overtime {
  time_factor: 0.3
}
time_control {
  c_denom: 20
  c_maxply: 40
  reserved_time: 1
}

Feb 24 20:38:51 !!!!! [13530] stderr: I0224 20:38:51.014569 13530 mcts_engine.cc:85] MCTSEngine: waiting all eval threads init

Feb 24 20:38:51 !!!!! [13530] stderr: I0224 20:38:51.174643 13531 trt_zero_model.cc:41] Added linear block of size 1478656
I0224 20:38:51.174666 13531 trt_zero_model.cc:41] Added linear block of size 1478656
I0224 20:38:51.174670 13531 trt_zero_model.cc:41] Added linear block of size 1478656
I0224 20:38:51.174674 13531 trt_zero_model.cc:41] Added linear block of size 1478656

Feb 24 20:38:52 !!!!! [13530] stderr: I0224 20:38:52.263502 13531 trt_zero_model.cc:97] tensorrt max batch size: 4
I0224 20:38:52.263536 13531 trt_zero_model.cc:108] tensorrt binding: inputs (19, 19, 17)
I0224 20:38:52.263551 13531 trt_zero_model.cc:108] tensorrt binding: policy (362, 1, 1)

Feb 24 20:38:52 !!!!! [13530] stderr: I0224 20:38:52.263564 13531 trt_zero_model.cc:108] tensorrt binding: value (1, 1, 1)

Feb 24 20:38:52 !!!!! [13530] stderr: I0224 20:38:52.271703 13531 mcts_engine.cc:359] EvalRoutine: init model done, global_step=639200

Feb 24 20:38:52 !!!!! [13530] stderr: I0224 20:38:52.271760 13530 mcts_engine.cc:87] MCTSEngine: all eval threads init done
I0224 20:38:52.271847 13530 mcts_main.cc:277] input cmd: boardsize 19
I0224 20:38:52.271909 13530 mcts_main.cc:286] = 
I0224 20:38:52.271982 13530 mcts_main.cc:277] input cmd: clear_board
I0224 20:38:52.272017 13530 mcts_main.cc:286] = 
I0224 20:38:52.271992 13540 mcts_engine.cc:922] DeleteRoutine: deleted 1 nodes, cost 0.00134ms
I0224 20:38:52.272045 13540 mcts_engine.cc:922] DeleteRoutine: deleted 1 nodes, cost 0.00035ms

Feb 24 20:38:52 !!!!! [13530] stderr: I0224 20:38:52.272053 13530 mcts_main.cc:277] input cmd: komi 7.5
I0224 20:38:52.272086 13530 mcts_main.cc:286] = 
I0224 20:38:52.272099 13530 mcts_main.cc:277] input cmd: play black q16
I0224 20:38:52.272351 13530 mcts_main.cc:87] Config hasn't changed
I0224 20:38:52.272388 13530 mcts_engine.cc:152] Move: pp

Feb 24 20:38:52 !!!!! [13530] stderr: I0224 20:38:52.272406 13540 mcts_engine.cc:922] DeleteRoutine: deleted 1 nodes, cost 0.00042ms
I0224 20:38:52.272413 13530 mcts_engine.cc:158] 1th move(b): pp, winrate=-nan%, N=0, Q=-nan, p=0.000000, v=nan, global_step=639200
1th move(b): pp, winrate=-nan%, N=0, Q=-nan, p=0.000000, v=nan, global_step=639200
I0224 20:38:52.272446 13530 mcts_main.cc:286] = 
I0224 20:38:52.272459 13530 mcts_main.cc:277] input cmd: showboard
E0224 20:38:52.272469 13530 mcts_main.cc:262] invalid op: showboard
I0224 20:38:52.272475 13530 mcts_main.cc:286] ? unknown command
I0224 20:38:52.272485 13530 mcts_main.cc:277] input cmd: genmove white
I0224 20:38:52.272640 13530 mcts_main.cc:87] Config hasn't changed

Feb 24 20:38:52   [13530] ? unknown command
Feb 24 20:38:52   [13530] 
Feb 24 20:38:52   [13530] Stopping bot
Feb 24 20:38:52   [Game 5073]  Move 1    gamedata     W   metaphysician splurgist :)  [19x19]    
Feb 24 20:38:52   [Game 5073]  Move 1    Starting new bot process [13547]
Feb 24 20:38:52   [Game 5073]  Move 1    State loading for new bot
Feb 24 20:38:52   [Game 5073]  Move 1    genmove white
Feb 24 20:38:52 !!!!! [13547] stderr: I0224 20:38:52.477252 13547 mcts_main.cc:75] load config succ: 
num_eval_threads: 1
num_search_threads: 8
max_search_tree_size: 400000000
max_children_per_node: 64
max_simulations_per_step: 3000
eval_batch_size: 4
eval_wait_batch_timeout_us: 100
model_config {
  train_dir: "/home/amd2018/PhoenixGo/ckpt/"
  enable_tensorrt: true
  tensorrt_model_path: "/home/amd2018/PhoenixGo/ckpt/zero.ckpt-20b-v1.FP32.PLAN"
}
gpu_list: "0"
c_puct: 2.5
virtual_loss: 1
enable_resign: true
v_resign: -0.8
dirichlet_noise_alpha: 0.03
dirichlet_noise_ratio: 0.25
policy_temperature: 0.67
early_stop {
  check_every_ms: 100
  sims_factor: 1
  sims_threshold: 2000
}
unstable_overtime {
  time_factor: 0.3
}
behind_overtime {
  time_factor: 0.3
}
time_control {
  c_denom: 20
  c_maxply: 40
  reserved_time: 1
}

Feb 24 20:38:52 !!!!! [13547] stderr: I0224 20:38:52.477983 13547 mcts_engine.cc:85] MCTSEngine: waiting all eval threads init

Feb 24 20:38:52 !!!!! [13547] stderr: I0224 20:38:52.637938 13548 trt_zero_model.cc:41] Added linear block of size 1478656
I0224 20:38:52.637962 13548 trt_zero_model.cc:41] Added linear block of size 1478656
I0224 20:38:52.637967 13548 trt_zero_model.cc:41] Added linear block of size 1478656
I0224 20:38:52.637970 13548 trt_zero_model.cc:41] Added linear block of size 1478656

Feb 24 20:38:53 !!!!! [13547] stderr: I0224 20:38:53.739930 13548 trt_zero_model.cc:97] tensorrt max batch size: 4
I0224 20:38:53.739964 13548 trt_zero_model.cc:108] tensorrt binding: inputs (19, 19, 17)
I0224 20:38:53.739989 13548 trt_zero_model.cc:108] tensorrt binding: policy (362, 1, 1)

Feb 24 20:38:53 !!!!! [13547] stderr: I0224 20:38:53.740005 13548 trt_zero_model.cc:108] tensorrt binding: value (1, 1, 1)

Feb 24 20:38:53 !!!!! [13547] stderr: I0224 20:38:53.748147 13548 mcts_engine.cc:359] EvalRoutine: init model done, global_step=639200

Feb 24 20:38:53 !!!!! [13547] stderr: I0224 20:38:53.748221 13547 mcts_engine.cc:87] MCTSEngine: all eval threads init done
I0224 20:38:53.748322 13547 mcts_main.cc:277] input cmd: boardsize 19
I0224 20:38:53.748412 13547 mcts_main.cc:286] = 
I0224 20:38:53.748473 13547 mcts_main.cc:277] input cmd: clear_board
I0224 20:38:53.748510 13547 mcts_main.cc:286] = 
I0224 20:38:53.748530 13547 mcts_main.cc:277] input cmd: komi 7.5
I0224 20:38:53.748522 13557 mcts_engine.cc:922] DeleteRoutine: deleted 1 nodes, cost 0.00122ms
I0224 20:38:53.748559 13557 mcts_engine.cc:922] DeleteRoutine: deleted 1 nodes, cost 0.00035ms
I0224 20:38:53.748566 13547 mcts_main.cc:286] = 
I0224 20:38:53.748585 13547 mcts_main.cc:277] input cmd: play black q16

Feb 24 20:38:53 !!!!! [13547] stderr: I0224 20:38:53.748888 13547 mcts_main.cc:87] Config hasn't changed
I0224 20:38:53.748939 13547 mcts_engine.cc:152] Move: pp
I0224 20:38:53.748975 13547 mcts_engine.cc:158] 1th move(b): pp, winrate=-nan%, N=0, Q=-nan, p=0.000000, v=nan, global_step=639200
I0224 20:38:53.748963 13557 mcts_engine.cc:922] DeleteRoutine: deleted 1 nodes, cost 0.00045ms
1th move(b): pp, winrate=-nan%, N=0, Q=-nan, p=0.000000, v=nan, global_step=639200
I0224 20:38:53.749013 13547 mcts_main.cc:286] = 
I0224 20:38:53.749032 13547 mcts_main.cc:277] input cmd: showboard
E0224 20:38:53.749050 13547 mcts_main.cc:262] invalid op: showboard
I0224 20:38:53.749058 13547 mcts_main.cc:286] ? unknown command
I0224 20:38:53.749073 13547 mcts_main.cc:277] input cmd: genmove white

Feb 24 20:38:53   [13547] ? unknown command
Feb 24 20:38:53   [13547] 
Feb 24 20:38:53   [13547] Stopping bot
Feb 24 20:38:53   [Game 5073]  Move 1    gamedata     W   metaphysician splurgist :)  [19x19]    
Feb 24 20:38:53   [Game 5073]  Move 1    Bot has crashed too many times, resigining game
Feb 24 20:38:53   [Game 5073]  Move 1    Failed to start the bot, can not make a move
Feb 24 20:38:53   [Game 5073]  Move 1    gamedata     W   metaphysician splurgist :)  [19x19]    
Feb 24 20:38:53   [Game 5073]  Move 1    Bot has crashed too many times, resigining game
Feb 24 20:38:53   [Game 5073]  Move 1    Failed to start the bot, can not make a move
Feb 24 20:38:54   [Game 5073]  Move 1    gamedata     W   metaphysician splurgist :)  [19x19]    
Feb 24 20:38:54   [Game 5073]  Move 1    Game over.   Result: B+R         L
Feb 24 20:38:54   [Game 5073]  Move 1    gamedata     W   metaphysician splurgist :)  [19x19]    
Feb 24 20:38:55   [Game 5073]  Move 1    Disconnecting from game.

```

very very nice !!

wonderingabout commented:
works as intended : 

https://beta.online-go.com/game/5075

windo commented:
@wonderingabout, ah yes, we were discussing the message but it slipped my mind, sorry!

wonderingabout commented:
@windo 

no problem, welcome :+1: 

remember that the messages are displayed in the wrong order, this is what i wanted to remind you, no problem for adding it :1st_place_medal:
